src/signature.rs

```diff
@@ -76,24 +80,15 @@ fn check_scalar(bytes: [u8; 32]) -> Result<Scalar, SignatureError> {
         return Err(InternalError::ScalarFormat.into());
     }
 
+    // You cannot do arithmetic with scalars construct with Scalar::from_bits. We only use this
+    // scalar for EdwardsPoint::vartime_double_scalar_mul_basepoint, which is an accepted usecase.
+    #[allow(deprecated)]
     Ok(Scalar::from_bits(bytes))
 }
 
 #[cfg(not(feature = "legacy_compatibility"))]
```

Comment:
```suggestion
#[cfg(not(feature = "legacy_compatibility"))]
#[allow(deprecated)]
```

Reply:
I'm confused, there's no deprecated code in the non-`legacy_compatibility` path. What does this macro do?

Reply:
Sorry, this is the wrong place.

I was trying to suggest where to add it to fix the test failure.

Reply:
It's failing because of the intentional deprecation warning on the `legacy_compat` version of `check_scalar()`. We can either remove that, letting ed- users not see the warning, or let them see the warning  and change some flags in the CI

Reply:
`allow(deprecated)` needs to be added here: https://github.com/dalek-cryptography/ed25519-dalek/blob/1cf99d7d06a11572f532871de583dc15c2cf79f4/src/signature.rs#L156

...or to `from_bytes`. That will fix the tests.

Reply:
Isn't that equivalent to just removing `#[deprecated(...)]` from `check_scalar()`? It's not a public function, and this is the only place it's used.

Reply:
Oh, well in that case there's not a lot of reason to have a `#[deprecated]` attribute. Maybe just make it clear in the comments?

Reply:
Makes sense. I'll do the latter tn